core/src/test/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilderTest.java

```diff
@@ -0,0 +1,222 @@
+
+        builder = new GeoBoundingBoxQueryBuilder(null);
+        builder.topLeft(0.0, 0.0);
+        builder.bottomRight(0.0, 0.0);
+        assertNotNull("null as fieldname should not be valid.", builder.validate());
+
+        // add remainder of interval, make interval exclusive
+        double invalidTop = randomFrom(invalidRandomDouble(MIN_LAT, MAX_LAT));
+        double invalidBottom = randomFrom(invalidRandomDouble(MIN_LAT, MAX_LAT));
+        double invalidLeft = randomFrom(invalidRandomDouble(MIN_LON, MAX_LON));
+        double invalidRight = randomFrom(invalidRandomDouble(MIN_LON, MAX_LON));
+
+        builder = new GeoBoundingBoxQueryBuilder("fieldname");
+        builder.topLeft(invalidTop, 0.0);
+        builder.bottomRight(0.0, 0.0);
+        assertNotNull("Validation should detect invalid top coordinate: " + invalidTop, builder.validate());
+
+        builder = new GeoBoundingBoxQueryBuilder("fieldname");
+        builder.topLeft(0.0, invalidLeft);
+        builder.bottomRight(0.0, 0.0);
+        assertNotNull("Validation should detect invalid left coordinate: " + invalidLeft, builder.validate());
+
+        builder = new GeoBoundingBoxQueryBuilder("fieldname");
+        builder.topLeft(0.0, 0.0);
+        builder.bottomRight(invalidBottom, 0.0);
+        assertNotNull("Validation should detect invalid bottom coordinate: " + invalidBottom, builder.validate());
+
+        builder = new GeoBoundingBoxQueryBuilder("fieldname");
+        builder.topLeft(0.0, 0.0);
+        builder.bottomRight(0.0, invalidRight);
+        assertNotNull("Validation should detect invalid right coordinate: " + invalidRight, builder.validate());
+
+        double bottom = randomDoubleBetween(MIN_LAT, MAX_LAT);
+        while (bottom == MAX_LAT) {
+            bottom = randomDoubleBetween(MIN_LAT, MAX_LAT);
+        }
+        double top = randomDoubleBetween(bottom, MAX_LAT);
+        double right = randomDoubleBetween(MIN_LON, MAX_LON);
+        while (right == MAX_LON) {
+            right = randomDoubleBetween(MIN_LON, MAX_LON);
+        }
+        double left = randomDoubleBetween(right, MAX_LON);
+        builder = new GeoBoundingBoxQueryBuilder("fieldname");
+        builder.topLeft(top, left);
+        builder.bottomRight(bottom, right);
+        assertNotNull("Validation should detect invalid right coordinate: left: " + left + " vs. right:" + right, builder.validate());
+
+        top = randomDoubleBetween(MIN_LAT, MAX_LAT, true, false);
+        bottom = randomDoubleBetween(top, MAX_LAT);
+        right = randomDoubleBetween(MIN_LON, MAX_LON, true, false);
+        left = randomDoubleBetween(right, MAX_LON);
+        builder = new GeoBoundingBoxQueryBuilder("fieldname");
+        builder.topLeft(top, left);
+        builder.bottomRight(bottom, right);
+        assertNotNull("Validation should detect invalid right coordinate: top: " + top + " vs. bottom: " + bottom, builder.validate());
```

Comment:
Also maybe simplify this test by generating a valid query and then switching topLeft/bottomRight? Shouldn't this cover all cases?